### ISSUE TITLE
Message button not working from dialer

### ISSUE THREAD

**Issue by lgexalter** [sspanak/tt9#306]
On the TCL Flip 2 there is an option in the dialer to send a message to a dialed number by clicking the soft left key. On the latest release of tt9 nothing happens when pressing the soft left key. The function only works when first pressing options (soft right key) then back, then afterwards pressing soft left for message.

**Comment by sspanak** [comment]
This could be related to #241.

**Comment by sspanak** [comment]
@lgexalter, could you please clarify what the soft right key does? I assume it opens up some context menu, which takes away the focus from the phone number field. Or does it do something else? My guess is the "send message" button is not working when the field is focused and you can keep typing.

**Comment by sspanak** [comment]
Um, I thought this your "OK" or some other confirmation key or maybe it isn't? 

If it is "OK", there are some issues with it, but they seem to have been fixed by Alex in #309.

**Comment by lgexalter** [comment]
@sspanak did not understand your last comment

**Comment by sspanak** [comment]
I thought the left key on your phone plays the role of an OK or ENTER key, call it whatever you want. We discovered a bug with how this key is handled, so that may be the reason you are not able to send messages.

It is also possible for this to be a SOFT_LEFT key (as per the documentation), but the dialer app could be remapping it virtually to ENTER. Android permits that, too. In this case, you would have the same problem.

And if it is a true SOFT_LEFT, well... no idea...

Have you assigned any special functionality to it in Settings -> Keypad -> Hotkeys? This would also prevent the key from doing its default action.

**Comment by lgexalter** [comment]
No, there is a separate OK/ENTER key and the SOFT_LEFT does not have any hotkey function. Pretty sure this issue was not present in older versions of tt9 (before #225 was fixed) the but I will double check that.

**Comment by sspanak** [comment]
@lgexalter, you may want to try the new v25.0. I fixed several cases of incorrect handling of key press and release events, that prevented the keys from working. If you have time, please let me know if the update fixes your problem, too.

**Comment by koshertek** [comment]
> @lgexalter, you may want to try the new v25.0. I fixed several cases of incorrect handling of key press and release events, that prevented some keys from working in various scenarios. If you have time, please let me know if the update fixes your problem, too.

Still broken

**Comment by sspanak** [comment]
Then I have no idea how to fix it. Sorry!

If anyone with a TCL Flip 2 is willing to debug this, you could try looking into the `KeyPadHandler` class. Find `onKeyDown()` and `onKeyUp()` and see what exactly they are doing. In order for the button to work properly, they must either return `false` or call the respective `super` method, but they are probably doing something else. Please, post the results here and hopefully, we will find a solution.

**Comment by sspanak** [comment]
@lgexalter, I think I've found the reason why this is happening. It is most likely the same as #300. It's just that the dialer is not detected properly, causing TT9 to run in the wrong mode and mishandle the Message key.

If you are willing to help, could you please follow the instructions in [this post](https://github.com/sspanak/tt9/issues/300#issuecomment-2058437762)? I need the Device Info and the package name of the dialer. 

Now, on the TCL Flip 2, it may be a bit tricky to copy the logs, because it probably does not have any messaging or notes app, where you can paste and review them, and you can't scroll to the end, where they are displayed in TT9, without a touch screen. If you can't figure it out, in the next version, I'll add a button to export the logs to a file, without any hassle.

**Comment by lgexalter** [comment]
> Out of curiosity, how did you get the package name when there were no logs? Did you read logcat yourself?

I happen to have known the package name. Double checked with `pm list packages dialer`.

> And final question, when you are dialing a number, is the TT9 status bar visible? I mean, do you see this:
>image
>?

No I do not

**Comment by lgexalter** [comment]
Sorry for the delay. Just tested v30.10 and the problem is not fixed

**Comment by sspanak** [comment]
My approach may be wrong, then. Just to confirm, could you please go again to the debugging screen, change the Keypad Handling Mode to "return false" or "super" and then go back to check if the message button has started working? 

By changing the mode, you will be able to type only numbers, so don't forget to restore to "normal" after testing.

**Comment by sspanak** [comment]
There seems to be no interest in this problem or it is not happening anymore. I am closing this issue for now.
